src/test/java/io/github/fbiville/nestedrelationquerysdn6/NestedRelationQuerySdn6ApplicationTests.java

```diff
@@ -14,65 +16,133 @@
     }
 
-    @BeforeEach
-    void prepare() {
+    @BeforeAll
+    void prepare(@Autowired ChangesetRepository repository) {
         try (Session session = driver.session()) {
             session.writeTransaction(tx -> {
                 tx.run("MATCH (n) DETACH DELETE n");
-                tx.run("CREATE (:Changeset:Hypercube {USER: 'some-user'})-[:CHANGESET_IN]->(ws:Workspace {STATE: 'some-state'}), " +
-                        "     (ws)-[:WORKSPACE_IN]->(:Hypercube:Work {STATE: 'some-other-state'})");
+                //tx.run("CREATE (:Workspace:Hypercube {STATE: 'some-xxx', ID : 1 , CANCELED : false})<-[:CHANGESET_IN {VALID : true}]-(:Changeset:Hypercube {USER: 'some-user', ID: 1, CANCELED : false})-[:CHANGESET_IN {VALID : true}]->(ws:Workspace:Hypercube {STATE: 'some-state', ID : 1 , CANCELED : false}), " +
+                tx.run("CREATE (:Changeset:Hypercube {USER: 'some-user', ID : 1, CANCELED : false})-[:CHANGESET_IN {VALID : true}]->(ws:Workspace:Hypercube {STATE: 'some-state', ID : 1 , CANCELED : false}), " +
+                        "     (ws)-[:WORKSPACE_IN]->(:Hypercube:Work {STATE: 'some-other-state', ID : 1 , CANCELED : false})");
                 return null;
             });
         }
     }
 
     @Test
-    void finds_changeset_by_id(@Autowired ChangesetRepository repository) {
+    void finds_changeset_by_id_rels_rels(@Autowired ChangesetRepository repository) {
         Long changesetId = findChangesetId("some-user");
 
-        Optional<Changeset> result = repository.findById(changesetId);
+        Optional<Changeset> result = repository.findByIdAndCanceledFalse(changesetId);
 
         assertThat(result).hasValueSatisfying((changeset) -> {
             assertThat(changeset.getId()).isEqualTo(changesetId);
             assertThat(changeset.getUser()).isEqualTo("some-user");
-            Map<Workspace, ChangesetIn> changesetRels = changeset.getChangesetIn();
+            Map<String, List<ChangesetIn>> changesetRels = changeset.getChangesetIn();
             assertThat(changesetRels).hasSize(1);
-            Workspace workspace = changesetRels.keySet().iterator().next();
+            String key = changesetRels.keySet().iterator().next();
+            Workspace workspace = (Workspace) changesetRels.get(key).get(0).getWorkspace();
             assertThat(workspace.getState()).isEqualTo("some-state");
-            Map<Work, WorkspaceIn> workspaceRels = workspace.getWorkspaceIn();
+            Map<String, List<WorkspaceIn>> workspaceRels = workspace.getWorkspaceIn();
             assertThat(workspaceRels).hasSize(1);
-            Work work = workspaceRels.keySet().iterator().next();
+            key = workspaceRels.keySet().iterator().next();
+            Work work = (Work) workspaceRels.get(key).get(0).getWork();
             assertThat(work.getState()).isEqualTo("some-other-state");
-        });
+        }); 
+    }
+    
+    @Test
+    void finds_changeset_by_id_rels_list(@Autowired ChangesetRepository repository) {
```

Comment:
I'm not sure I understand what these tests are checking that the initial test does not. Would you mind elaborating on this?